application/database/database_routes.py

```diff
@@ -55,8 +60,21 @@ def get_recent_location():
 def list_location():
   """
   list all locations
+  optionally support filters
   """
-  table_data = Entities.query.order_by(Entities.record_id).all()
+
+  filter_latitude = request.args.get('source_latitude')
+  filter_longitude = request.args.get('source_longitude')
+  filter_distance = request.args.get('distance') # miles
+
+  if filter_latitude and filter_longitude and filter_distance:
+    try:
+      table_data = Entities.query.filter(text("POINT({:f}, {:f}) <@> POINT(location_longitude, location_latitude) < {:f}".format(float(filter_longitude), float(filter_latitude), float(filter_distance)))).order_by(Entities.record_id)
```

Comment:
we're not listing these points on the front end...i recommend removing the sorting call in order to minimize performance overhead. this isn't a huge issue now, but will become an issue shortly as the number of records doubles and triples in the next week or so

Reply:
given it's the primary key it's all in memory so shouldn't actually make a material difference:

>    "entities_pkey" PRIMARY KEY, btree (record_id)
>    "ix_entities_location_id" btree (location_id)

if we get to a point where we need to optimize queries I'll be happy ;) sorting is redundant tho so I will kill it

Reply:
done